### PR TITLE
Feature/send order to kitchen

### LINKED ISSUE #35: Send menu dishes to kitchen

### PR DESCRIPTION
**Pull request by jcmexdev**
Features:

- Change order status from preparing to waiting
- Validate order totalDishes before update status
- Validate that the order status is preparing to update
- Update totalDishes in order when add or remove a dish

### CODE REVIEW

File: src/modules/order-detail/order-detail.service.ts [hunk 1]
```diff
@@ -45,6 +49,9 @@ export class OrderDetailService {
     orderDetail.price = menuDish.price;
     await orderDetail.save();
 
+    order.totalDishes++;
```
Comment (jlazo) [inline src/modules/order-detail/order-detail.service.ts]: Also You must to set the totalPrice in order

File: src/modules/order-detail/order-detail.service.ts [hunk 2]
```diff
@@ -59,8 +66,16 @@ export class OrderDetailService {
   // TODO add documentation api
   async remove(id: number) {
     const detail = await this._orderDetailRepository.findOne(id);
-    if (!detail) throw new ConflictException('order_detail_not_found');
+
+    if (!detail) throw new NotFoundException('order_detail_not_found');
+
+    const order = await this._orderRepository.findOne(detail.orderId);
+
+    order.totalDishes--;
```
Comment (jlazo) [inline src/modules/order-detail/order-detail.service.ts]: Also you must update the price in order

File: src/modules/order-detail/order-detail.service.ts [hunk 3]
```diff
@@ -59,8 +66,16 @@ export class OrderDetailService {
...
+    order.save();
+
     detail.remove();
```
Comment (jlazo) [inline src/modules/order-detail/order-detail.service.ts]: if the detail was removed so you update the order

File: src/modules/order/order.service.ts [hunk 4]
```diff
@@ -53,4 +56,20 @@ export class OrderService {
...
+
+    if (!order) throw new NotFoundException('order_was_not_found');
+
+    if (order.status != OrderStatus.ORDERING)
+      throw new ConflictException('the_order_was_already_shipped_previously');
+
+    if (order.totalDishes < 1)
+      throw new ConflictException('it_cant_ship_because_order_is_empty');
+
+    order.status = OrderStatus.WAITING;
```
Comment (jlazo) [inline src/modules/order/order.service.ts:74]: Where do you update the status of order details?

Comment (jcmexdev) [reply src/modules/order/order.service.ts:74]: It is in https://github.com/Swat-FastOrder/Backend/pull/66

**Review by jlazo — APPROVED** [review]

File: src/modules/order/order.service.ts [hunk 5]
```diff
@@ -4,18 +4,21 @@ import {
...
 import { Order } from './order.entity';
 import { OrderRepository } from './order.repository';
+import { OrderStatus } from './order.status.enum';
 
 @Injectable()
 export class OrderService {
   constructor(
     private readonly _orderRepository: OrderRepository,
     private readonly _tableRepository: TableRepository,
+    private readonly _orderDetails: OrderDetailRepository,
```
Comment (AlanDavd) [inline src/modules/order/order.service.ts]: Why we are not using this variable?

**Review by AlanDavd — APPROVED** [review]
LGTM

**Pull request merged**